### PR TITLE
fixed product data and request tests

### PR DESCRIPTION
**Pull request by programjon**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 /node_modules
 /coverage
+.idea/
```
Comment (ptte) [inline .gitignore:3]: what is .idea?

File: index.js [hunk 2]
```diff
@@ -239,6 +240,59 @@ var extract = function (url, html, res) {
     });
   });
 
+  /*
+  comment this!
+   */
+  if (meta.parse_products === 1) {
+
+    var supportedURLs = [];
+    request('http://api.twotap.com/v1.0/supported_sites', function (error, response, body) {
```
Comment (ptte) [inline index.js]: Is this a very often changing list?

File: index.js [hunk 3]
```diff
@@ -239,6 +240,59 @@ var extract = function (url, html, res) {
...
+        $('a').each(function () {
+          var link = $(this);
+          var href = link.attr('href');
+          var thisurl = urlModule.parse(href).hostname;
+          if (supportedURLs.indexOf(thisurl) !== -1) {
+            data.hrefs.push(href);
+          }
+        });
+        if (data.hrefs.length > 0) {
+          var public_token = '9b8ca38064638f40ec04aec7ae3cc9';
```
Comment (ptte) [inline index.js]: is this our private secret token for the twotap api?

File: index.js [hunk 4]
```diff
@@ -270,7 +324,10 @@ var fetch = function (url, meta, options, callback) {
       }
     }
     if (!err && !result.redirect) {
-      result.data = extract(url, body, res);
+      var testmetabad = [];
+      testmetabad.parse_products = 9;
```
Comment (ptte) [inline index.js]: curious, what is this?

File: index.js [hunk 5]
```diff
@@ -239,6 +240,12 @@ var extract = function (url, html, res) {
     });
   });
 
+  $('a').each(function () {
+    var link = $(this);
+    var href = link.attr('href');
```
Comment (ptte) [inline index.js]: Seems easier to just do `$(this).attr('href')`

File: index.js [hunk 6]
```diff
@@ -22,7 +21,7 @@ var ogTypes = [
   'music',
   'article',
   'book',
-  'profile',
+  'profile'
```
Comment (ptte) [inline index.js]: trailing commas are good :D

**Comment by ptte** [comment]
this is starting to look nice and clean :)

some small fixes, and squash up the commits to one, and we're good to go

File: .gitignore (hunk 1 above)
Comment (programjon) [inline .gitignore:3]: webstorm/jetbrains IDE settings stuff

**Comment by ptte** [comment]
lgtm! :)

**Pull request merged**